### DISCUSSION TITLE
ETC support

### DISCUSSION THREAD

**Discussion by BBB1331** [coinfoundry/miningcore#1415] [General]
hi when will be added etc ?

**Comment by oliverw** [comment]
> hi when will be added etc ?

As soon as your PR is in. 😦

**Comment by Silur** [comment]
I believe https://github.com/oliverw/miningcore/blob/506f32134ef45daea16b0f9f15859a9583a6c31a/src/Miningcore/Crypto/Hashing/Ethash/Dag.cs#L87 this is the only part where the hashing needs to be replaced with a call to https://github.com/eth-classic/etchash instad of "stock" ethash, otherwise having an etc fullnode as upstream with an ethereum config works. Only shares break because of the different hashing.

**Reply by BBB1331** [reply]
what u mean ?it can work?

**Comment by Konstantin35** [comment]
[libetchash.zip](https://github.com/oliverw/miningcore/files/10327949/libetchash.zip)

**Reply by BBB1331** [reply]
brat kak eto ustanovit ?vi mojete chuchut podskazat?

**Reply by Konstantin35** [reply]
Это можно использовать аналогично libethhash, но длина эпохи должна быть 60000

**Reply by BBB1331** [reply]
ya mnoga kopal nu nichevo ne poluchilos)))

**Reply by BBB1331** [reply]
THX Worked)))

**Reply by blackmennewstyle** [reply]
You will probably need to create `Crypto/Hashing/Etchash` based on `Crypto/Hashing/Ethash`and also `Native/Etchash` based on `Native/Ethash`. And finally, tell `EthereumJobManager` to use `Etchash` instead of `Ethash`. it's the more easy dirty implementation that i can see :)

**Reply by BBB1331** [reply]
i finished now working .i didn`t catch block but all miners working thx to Konstantin35  and oliverw

**Reply by blackmennewstyle** [reply]
Nice :)
Will you make a PR later on?

**Reply by RulezZzOr** [reply]
could anyone share the core, or update the release?

**Reply by MiningCryptoLive** [reply]
john4z has it on GitHub. I am using it and it works great

**Reply by MiningCryptoLive** [reply]
https://github.com/jon4hz/miningcore-foss and checkout ravencoin branch for etc and ravencoin.

**Reply by blackmennewstyle** [reply]
Actually the `dev` branch has also `ETC` enabled.
It's probably worth to wait a bit since the final version i believe will have an implementation allowing users to choose between `full DAG cache` and `ligth DAG cache`.

There are some non negligible latency differences between both options but they both have their advantages depending on your server architecture.
